File: Engine/Src/Protocol/Protocol/GenericRPCSendAdapter.cpp
////////////////////////////////////////////////////////////////////////////////
// 
// CopyRight (c) The Braves
// 
// Author : Generated
// 
// Description : Generic Message debug implementations
// 
////////////////////////////////////////////////////////////////////////////////


#include "SFProtocolPCH.h"
#include "SFTypedefs.h"
#include "Net/SFNetDef.h"
#include "Net/SFMessage.h"
#include "Net/SFMessageEndpoint.h"
#include "Actor/Movement/SFActorMovement.h"
#include "Protocol/GenericRPCSendAdapter.h"
#include "Protocol/GenericMessageID.h"
#include "Generic_generated.h"



namespace SF
{
 	// Cmd: Generic failure message
	Result GenericRPCSendAdapter::GenericFailureCmd( const TransactionID &InTransactionID )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		SF::Flat::Generic::GenericFailureCmdBuilder _builder(fbb);
		flatbuffers::Offset<SF::Flat::Generic::GenericFailureCmd> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, ResultCode::SUCCESS, Message::Generic::MID_GenericFailureCmd, fbb));


		return hr;
	}; // Result GenericRPCSendAdapter::GenericFailureCmd( const TransactionID &InTransactionID )
	// Cmd: Generic transaction result
	Result GenericRPCSendAdapter::GenericTransactionCmd( const TransactionID &InTransactionID )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		SF::Flat::Generic::GenericTransactionCmdBuilder _builder(fbb);
		flatbuffers::Offset<SF::Flat::Generic::GenericTransactionCmd> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, ResultCode::SUCCESS, Message::Generic::MID_GenericTransactionCmd, fbb));


		return hr;
	}; // Result GenericRPCSendAdapter::GenericTransactionCmd( const TransactionID &InTransactionID )
	// C2S: Client heartbeat
	Result GenericRPCSendAdapter::HeartbeatC2SEvt(  )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		SF::Flat::Generic::HeartbeatC2SEvtBuilder _builder(fbb);
		flatbuffers::Offset<SF::Flat::Generic::HeartbeatC2SEvt> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(TransactionID(), ResultCode::SUCCESS, Message::Generic::MID_HeartbeatC2SEvt, fbb));


		return hr;
	}; // Result GenericRPCSendAdapter::HeartbeatC2SEvt(  )
	// Cmd: Post log data through gateway
	Result GenericRPCSendAdapter::PostLogDataCmd( const TransactionID &InTransactionID, const char* InDestLogChannel, const Array<uint8_t>& InMessageData )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		auto DestLogChannelOffset = SF::Flat::Helper::CreateString(fbb, InDestLogChannel);
		auto MessageDataOffset = SF::Flat::Helper::Createuint8Vector(fbb, InMessageData);
		SF::Flat::Generic::PostLogDataCmdBuilder _builder(fbb);
		_builder.add_dest_log_channel(DestLogChannelOffset);
		_builder.add_message_data(MessageDataOffset);
		flatbuffers::Offset<SF::Flat::Generic::PostLogDataCmd> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, ResultCode::SUCCESS, Message::Generic::MID_PostLogDataCmd, fbb));


		return hr;
	}; // Result GenericRPCSendAdapter::PostLogDataCmd( const TransactionID &InTransactionID, const char* InDestLogChannel, const Array<uint8_t>& InMessageData )


	// Cmd: Generic failure message
	Result GenericSvrRPCSendAdapter::GenericFailureRes( const TransactionID &InTransactionID, const Result &InResult )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		SF::Flat::Generic::GenericFailureResBuilder _builder(fbb);
		flatbuffers::Offset<SF::Flat::Generic::GenericFailureRes> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, InResult, Message::Generic::MID_GenericFailureRes, fbb));


		return hr;
	}; // Result GenericSvrRPCSendAdapter::GenericFailureRes( const TransactionID &InTransactionID, const Result &InResult )
	// Cmd: Generic transaction result
	Result GenericSvrRPCSendAdapter::GenericTransactionRes( const TransactionID &InTransactionID, const Result &InResult, const TransactionID &InFinishedTransaction, const char* InSignature )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		auto FinishedTransactionOffset = SF::Flat::Helper::CreateTransactionID(fbb, InFinishedTransaction);
		auto SignatureOffset = SF::Flat::Helper::CreateString(fbb, InSignature);
		SF::Flat::Generic::GenericTransactionResBuilder _builder(fbb);
		_builder.add_finished_transaction(FinishedTransactionOffset);
		_builder.add_signature(SignatureOffset);
		flatbuffers::Offset<SF::Flat::Generic::GenericTransactionRes> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, InResult, Message::Generic::MID_GenericTransactionRes, fbb));


		return hr;
	}; // Result GenericSvrRPCSendAdapter::GenericTransactionRes( const TransactionID &InTransactionID, const Result &InResult, const TransactionID &InFinishedTransaction, const char* InSignature )
	// Cmd: Post log data through gateway
	Result GenericSvrRPCSendAdapter::PostLogDataRes( const TransactionID &InTransactionID, const Result &InResult )
	{
 		Result hr;

		protocolCheckPtr(m_Endpoint);

		flatbuffers::FlatBufferBuilder& fbb = GetBuilderForNew();
		SF::Flat::Generic::PostLogDataResBuilder _builder(fbb);
		flatbuffers::Offset<SF::Flat::Generic::PostLogDataRes> packetOffset = _builder.Finish();
		fbb.Finish(packetOffset);

		protocolCheck(Send(InTransactionID, InResult, Message::Generic::MID_PostLogDataRes, fbb));


		return hr;
	}; // Result GenericSvrRPCSendAdapter::PostLogDataRes( const TransactionID &InTransactionID, const Result &InResult )


}; // namespace SF

File: Engine/Src/SFProtocolSharp/Protocol/GenericRPCSendAdapter.cs
////////////////////////////////////////////////////////////////////////////////
// 
// CopyRight (c) StromForge
// 
// Author : Generated
// 
// Description : Generic Network RPC send adapter
// 
////////////////////////////////////////////////////////////////////////////////


using System;
using System.Diagnostics;
using SF;
using SF.Flat;
using SF.Flat.Generic;
#nullable enable



namespace SF.Net
{
 

	public class GenericRPCSendAdapter : RPCAdapter
	{
 


		public  GenericRPCSendAdapter()
		{
 		} // public  GenericRPCSendAdapter()
		public  GenericRPCSendAdapter( SF.IEndpoint endpoint ) : base(endpoint)
		{
 		} // public  GenericRPCSendAdapter( SF.IEndpoint endpoint ) : base(endpoint)

		// Cmd: Generic failure message
		public Result  GenericFailureCmd( SF.TransactionID InTransactionID, Action<SFMessage>? callback = null )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			SF.Flat.Generic.GenericFailureCmd.StartGenericFailureCmd(builder);
			var packetOffset = SF.Flat.Generic.GenericFailureCmd.EndGenericFailureCmd(builder);
			result = SendMessage(MessageIDGeneric.GenericFailureCmd, builder, packetOffset.Value, transactionId:InTransactionID , callback:callback);
			return result;
		} // public Result  GenericFailureCmd( SF.TransactionID InTransactionID, Action<SFMessage>? callback = null )

		// Cmd: Generic transaction result
		public Result  GenericTransactionCmd( SF.TransactionID InTransactionID, Action<SFMessage>? callback = null )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			SF.Flat.Generic.GenericTransactionCmd.StartGenericTransactionCmd(builder);
			var packetOffset = SF.Flat.Generic.GenericTransactionCmd.EndGenericTransactionCmd(builder);
			result = SendMessage(MessageIDGeneric.GenericTransactionCmd, builder, packetOffset.Value, transactionId:InTransactionID , callback:callback);
			return result;
		} // public Result  GenericTransactionCmd( SF.TransactionID InTransactionID, Action<SFMessage>? callback = null )

		// C2S: Client heartbeat
		public Result  HeartbeatC2SEvt(  )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			SF.Flat.Generic.HeartbeatC2SEvt.StartHeartbeatC2SEvt(builder);
			var packetOffset = SF.Flat.Generic.HeartbeatC2SEvt.EndHeartbeatC2SEvt(builder);
			result = SendMessage(MessageIDGeneric.HeartbeatC2SEvt, builder, packetOffset.Value);
			return result;
		} // public Result  HeartbeatC2SEvt(  )


		// Cmd: Post log data through gateway
		public Result  PostLogDataCmd( SF.TransactionID InTransactionID, System.String InDestLogChannel, System.Byte[] InMessageData, Action<SFMessage>? callback = null )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			var DestLogChannelOffset = builder.CreateString(InDestLogChannel);
			var MessageDataOffset = builder.Createuint8Vector(InMessageData);
			SF.Flat.Generic.PostLogDataCmd.StartPostLogDataCmd(builder);
			SF.Flat.Generic.PostLogDataCmd.AddDestLogChannel(builder, DestLogChannelOffset);
			SF.Flat.Generic.PostLogDataCmd.AddMessageData(builder, MessageDataOffset);
			var packetOffset = SF.Flat.Generic.PostLogDataCmd.EndPostLogDataCmd(builder);
			result = SendMessage(MessageIDGeneric.PostLogDataCmd, builder, packetOffset.Value, transactionId:InTransactionID , callback:callback);
			return result;
		} // public Result  PostLogDataCmd( SF.TransactionID InTransactionID, System.String InDestLogChannel, System.Byte[] InMessageData, Action<SFMessage>? callback = null )

	}; // public class GenericRPCSendAdapter : RPCAdapter
	public class GenericSvrRPCSendAdapter : RPCAdapter
	{
 


		public  GenericSvrRPCSendAdapter()
		{
 		} // public  GenericSvrRPCSendAdapter()
		public  GenericSvrRPCSendAdapter( SF.IEndpoint endpoint ) : base(endpoint)
		{
 		} // public  GenericSvrRPCSendAdapter( SF.IEndpoint endpoint ) : base(endpoint)

		// Cmd: Generic failure message
		public Result  GenericFailureRes( SF.TransactionID InTransactionID, SF.Result InResult )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			SF.Flat.Generic.GenericFailureRes.StartGenericFailureRes(builder);
			var packetOffset = SF.Flat.Generic.GenericFailureRes.EndGenericFailureRes(builder);
			result = SendMessage(MessageIDGeneric.GenericFailureRes, builder, packetOffset.Value, transactionId:InTransactionID, result:InResult);
			return result;
		} // public Result  GenericFailureRes( SF.TransactionID InTransactionID, SF.Result InResult )


		// Cmd: Generic transaction result
		public Result  GenericTransactionRes( SF.TransactionID InTransactionID, SF.Result InResult, SF.TransactionID InFinishedTransaction, System.String InSignature )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			var FinishedTransactionOffset = builder.CreateTransactionID(InFinishedTransaction);
			var SignatureOffset = builder.CreateString(InSignature);
			SF.Flat.Generic.GenericTransactionRes.StartGenericTransactionRes(builder);
			SF.Flat.Generic.GenericTransactionRes.AddFinishedTransaction(builder, FinishedTransactionOffset);
			SF.Flat.Generic.GenericTransactionRes.AddSignature(builder, SignatureOffset);
			var packetOffset = SF.Flat.Generic.GenericTransactionRes.EndGenericTransactionRes(builder);
			result = SendMessage(MessageIDGeneric.GenericTransactionRes, builder, packetOffset.Value, transactionId:InTransactionID, result:InResult);
			return result;
		} // public Result  GenericTransactionRes( SF.TransactionID InTransactionID, SF.Result InResult, SF.TransactionID InFinishedTransaction, System.String InSignature )


		// Cmd: Post log data through gateway
		public Result  PostLogDataRes( SF.TransactionID InTransactionID, SF.Result InResult )
		{
 			if (Endpoint == null) return ResultCode.IO_NOT_CONNECTED;
			Result result = ResultCode.SUCCESS;
			var builder = new Google.FlatBuffers.FlatBufferBuilder(1024);
			SF.Flat.Generic.PostLogDataRes.StartPostLogDataRes(builder);
			var packetOffset = SF.Flat.Generic.PostLogDataRes.EndPostLogDataRes(builder);
			result = SendMessage(MessageIDGeneric.PostLogDataRes, builder, packetOffset.Value, transactionId:InTransactionID, result:InResult);
			return result;
		} // public Result  PostLogDataRes( SF.TransactionID InTransactionID, SF.Result InResult )


	}; // public class GenericSvrRPCSendAdapter : RPCAdapter




} // namespace SF.Net
#nullable restore


